src/rapidfuzz/process_cpp_impl.pyx

```diff
@@ -1216,14 +1217,15 @@ def extract(query, choices, *, scorer=WRatio, processor=None, limit=5, score_cut
         return []
 
     try:
-        if limit is None or limit > len(choices):
-            limit = len(choices)
+        choices_len = <int64_t>len(choices)
     except TypeError:
         # handle generators. In Theory we could retrieve the length later on while
         # preprocessing the choices, but this is good enough for now
         choices = list(choices)
-        if limit is None or limit > len(choices):
-            limit = len(choices)
+        choices_len = <int64_t>len(choices)
+
+    if limit is None or limit > choices_len:
+        limit = choices_len
 
     c_limit = limit
```

Comment:
It's unnecessary to convert `choices_len` back to a Python Object here. This could probably be something like:

```suggestion
    c_limit = choices_len
    if limit is not None:
        c_limit = min(c_limit, <int64_t>limit)
```